### PR TITLE
Pivotal #ID 176479473:  Migrate biostudies test to TestContainers

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/176479473

### CODE REVIEW

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/integration/MongoDbReposConfig.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import ac.uk.ebi.biostd.persistence.doc.service.SubmissionMongoMetaQueryService
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Import
+import org.springframework.data.mongodb.core.MongoTemplate
+
+@Configuration
+@Import(MongoDbConfig::class)
+@ConditionalOnProperty(prefix = "app.persistence", name = ["enabledMongo"])
```
Comment (jhoanmanuelms) [inline submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/integration/MongoDbReposConfig.kt:17]: As I suggested in the previous PR. The property name should be either `enableMongo` or `mongoEnabled`.

Also, we need to update the file `application-local.yml` with the new properties.

File: submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionDocDataRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    @Autowired
+    lateinit var submissionMongoRepository: SubmissionMongoRepository
+
+    @Nested
+    inner class UpdateStatus {
+        @Test
+        fun `successful status update`() {
+            submissionMongoRepository.save(testDocSubmission("accNo1", 1, PROCESSING))
+
+            testInstance.updateStatus(PROCESSED, "accNo1", 1)
```
Comment (jhoanmanuelms) [inline submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionDocDataRepositoryTest.kt:43]: I think using the named `version` parameter would add more clarity to the tests

**Review by jhoanmanuelms — COMMENTED** [review]
I left some comments

File: submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionDocDataRepositoryTest.kt (hunk 2 above)
Comment (Juan-EBI) [reply submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/db/data/SubmissionDocDataRepositoryTest.kt:43]: this is really sugetive one may same the same about status and processes status not sure what convention of logic conclusion we can arrive to add parameters names or not

**Review by jhoanmanuelms — APPROVED** [review]
LGTM

**Pull request merged**